ktor-server/ktor-server-plugins/ktor-server-webjars/jvm/src/io/ktor/server/webjars/Webjars.kt

```diff
@@ -28,6 +37,39 @@ public class WebjarsConfig {
                 }
             }
         }
+
+    /**
+     * Specifies a value for [HttpHeaders.LastModified] to be used in the response.
+     * By default, it is the time when this [Application] instance started.
+     * Return `null` from this block to omit the header.
+     *
+     * Note: for this property to work, you need to install the [ConditionalHeaders] plugin.
+     */
+    public fun lastModified(block: (WebJarInfo) -> GMTDate?) {
+        lastModifiedExtractor = block
+    }
+
+    /**
+     * Specifies a value for [HttpHeaders.ETag] to be used in the response.
+     * By default, it is the WebJar version.
+     * Return `null` from this block to omit the header.
+     *
+     * Note: for this property to work, you need to install the [ConditionalHeaders] plugin.
+     */
+    public fun etag(block: (WebJarInfo) -> String?) {
+        etagExtractor = block
+    }
+
+    /**
+     * Specifies a value for [HttpHeaders.CacheControl] to be used in the response.
+     * By default, it is 90 days.
+     * Return `null` from this block to omit the header.
+     *
+     * Note: for this property to work, you need to install the [CachingHeaders] plugin.
```

Comment:
Can we install `CachingHeaders` and `ConditionalHeaders` with the `Webjars` plugin by default? Maybe we can add an option in the `Webjars` config to omit them then? I'm not sure it is convenient to install another plugin to be able to work with property from our plugin

Reply:
They are in different modules, so no, it's not easy. Maybe we can check dependency in runtime, but it won't help much, since the users will need to add the dependency anywat.